### ISSUE TITLE
Disable kicking AI BOTS on player join.

### ISSUE THREAD

**Issue by Martin-Veskilt** [CnCNet/xna-cncnet-client#157]
This is an issue to hosts who set up their game to play comp stomps or any games vs AI and the flood of joining players kick the AI BOTS. The hosted lobby should just lock automatically like when all player slots are full. 

This issue is affected by : [https://github.com/CnCNet/xna-cncnet-client/issues/156](url)

**Comment by Rampastring** [comment]
I think this comes down to personal preference. 

In DTA some people set up lobbies for primarily comp stomp, but might play PvP if enough people join. Sometimes there's a situation where a 4v4 comp stomp has been set up, but a player joins and people decide to play 5-player FFA instead. That couldn't happen if the game was locked automatically when the remaining player slots are filled with AIs.

I can see it being a problem in the busier games, but the game hosts there could just pick a smaller limit for maximum players (like 4 instead of 8) when hosting a game if they want to play comp-stomp.

**Comment by Martin-Veskilt** [comment]
> I think this comes down to personal preference.

Yes, this is true but the issue lies in the client behaviour being automated for a single preference. An option to disable this kind of behaviour in-game Options > CnCNet would give players a choice.

> Sometimes there's a situation where a 4v4 comp stomp has been set up, but a player joins and people decide to play 5-player FFA instead.

These same players can decide to kick the 5th player or even go for a 3v3 comp stomp, as that was their original plan. It's difficult to predict player preference or choice so automating for a single usage pattern doesn't make much sense. It's best to leave these decisions up to the players seeing as they can message each other to open games and remove bots manually based on their decisions or circumstances. 

> I can see it being a problem in YR and MO, but the game hosts there could just pick a smaller limit for maximum players (like 4 instead of 8) when hosting a game if they want to play comp-stomp.

This wouldn't change the issue at all, even if I would choose a 6 player limit for a 3v3 comp-stop the additional joining players that flood in will automatically kick all the AI bots.
